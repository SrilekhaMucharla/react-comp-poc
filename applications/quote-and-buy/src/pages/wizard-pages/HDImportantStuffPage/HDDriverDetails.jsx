import React from 'react';
import PropTypes from 'prop-types';
import { Row, Col } from 'react-bootstrap';
import * as messages from './HDQuoteDriverTable.messages';


const HDDriverDetails = ({
    driver: {
        dateOfBirth, occupation, drivingLicence, accidence, convictions
    },
}) => {
    const entryColsProps = { xs: 12, lg: 6 };
    const keyColsProps = { xs: 6 };
    const valueColsProps = { xs: 6 };
    return (
        <Row className="driver-details-card">
            <Col {...entryColsProps}>
                <Row>
                    <Col {...keyColsProps} className="col-xl-5 col-xxl-4 driver-details-card__key">{messages.dateOfBirth}</Col>
                    <Col {...valueColsProps} className="col-xl-7 col-xxl-8 driver-details-card__value">{dateOfBirth}</Col>
                </Row>
            </Col>
            <Col {...entryColsProps}>
                <Row>
                    <Col {...keyColsProps} className="col-xxl-6 driver-details-card__key">{messages.accidence}</Col>
                    <Col {...valueColsProps} className="col-xxl-6 driver-details-card__value">{accidence}</Col>
                </Row>
            </Col>
            <Col {...entryColsProps}>
                <Row>
                    <Col {...keyColsProps} className="col-xl-5 col-xxl-4 driver-details-card__key">{messages.occupation}</Col>
                    <Col {...valueColsProps} className="col-xl-7 driver-details-card__value">{occupation}</Col>
                </Row>
            </Col>
            {convictions && (
                <Col {...entryColsProps}>
                    <Row>
                        <Col {...keyColsProps} className="col-xxl-6 driver-details-card__key">{messages.convictions}</Col>
                        <Col {...valueColsProps} className="col-xxl-6 driver-details-card__value">{convictions}</Col>
                    </Row>
                </Col>
            )}
            <Col {...entryColsProps}>
                <Row>
                    <Col {...keyColsProps} className="col-xl-5 col-xxl-4 driver-details-card__key">{messages.drivingLicence}</Col>
                    <Col {...valueColsProps} className="col-xl-7 col-xxl-8 driver-details-card__value">{drivingLicence}</Col>
                </Row>
            </Col>
        </Row>
    );
};

HDDriverDetails.propTypes = {
    driver: PropTypes.shape({
        dateOfBirth: PropTypes.string.isRequired,
        occupation: PropTypes.string.isRequired,
        drivingLicence: PropTypes.string.isRequired,
        accidence: PropTypes.string.isRequired,
        convictions: PropTypes.string.isRequired,
    }).isRequired,
};

export default HDDriverDetails;
