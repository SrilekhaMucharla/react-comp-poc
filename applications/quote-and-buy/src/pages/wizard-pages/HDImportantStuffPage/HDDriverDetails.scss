.driver-details-card {
    padding: 30px 5px;
    @include media-breakpoint-up(md) {
        padding: 30px 15px;
    }
    & > div + div {
        margin-top: 10px;
        @include media-breakpoint-up(lg) {
            margin-top: 0;
        }
    }
    & > div + div + div {
        @include media-breakpoint-up(lg) {
            margin-top: 10px;
        }
    }
}

.driver-details-card__key {
    color: $info-color;
}

.driver-details-card__value {
    color: $app-theme;
}
