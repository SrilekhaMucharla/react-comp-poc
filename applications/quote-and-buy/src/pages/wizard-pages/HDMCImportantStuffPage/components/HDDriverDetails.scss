.mc-driver-details-card {
    padding: 30px 5px;
    @include media-breakpoint-up(md) {
        padding: 30px 15px;
    }
    & > div + div {
        margin-top: 10px;
        @include media-breakpoint-up(lg) {
            margin-top: 0;
        }
    }
    & > div + div + div {
        @include media-breakpoint-up(lg) {
            margin-top: 10px;
        }
    }
}

.mc-driver-details-card__key {
    color: $info-color;
}

.mc-driver-details-card__value {
    color: $app-theme;
    padding-right: 20px;
    .custom-padding-right {
        padding-right: 40px;
    }
}
