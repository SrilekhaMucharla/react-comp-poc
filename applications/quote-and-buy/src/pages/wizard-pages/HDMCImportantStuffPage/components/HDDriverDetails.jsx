/* eslint-disable no-nested-ternary */
import React, { useContext } from 'react';
import PropTypes from 'prop-types';
import { Row, Col } from 'react-bootstrap';
import { useSelector } from 'react-redux';
import { HDPlaceholderWithHeader } from 'hastings-components';
import dayjs from 'dayjs';
import driverAddedIcon from '../../../../assets/images/icons/Icons_Driver-added.svg';
import * as messages from './HDDriverDetails.messages';
import { TranslatorContext } from '../../../../integration/TranslatorContext';

const HDDriverDetails = ({ driver }) => {
    const translator = useContext(TranslatorContext);
    const driverLicenceDetails = useSelector((state) => state.wizardState.app.driverLicenceDetails);

    const keyColsProps = { xs: 6 };
    const valueColsProps = { xs: 6 };
    const columnKeySize = 'col-xl-5';
    const columnValueSize = 'col-xl-7';
    const columnEqualSize = 'col-xl-6';

    let licenceValidated = false;
    const licenceDetails = driverLicenceDetails && driverLicenceDetails.filter((driverLicenceDetail) => (
        driverLicenceDetail && driverLicenceDetail.displayName === driver.name
        && driverLicenceDetail.dob && driver.dateOfBirth
        && driverLicenceDetail.dob.getTime() === driver.dateOfBirth.getTime()));
    if (licenceDetails && licenceDetails[0]) {
        licenceValidated = licenceDetails[0].licenceSuccessfulValidated;
    }

    return (
        <HDPlaceholderWithHeader
            className="margin-top-lg"
            key={driver.id}
            title={(
                <span>
                    {driver.name}
                    <span className="font-regular">{driver.role && ` (${driver.role})`}</span>
                </span>
            )}
            icon={<img className="hd-driver-details-header-icon" src={driverAddedIcon} alt="Added driver icon" />}
        >
            <Row className="mc-driver-details-card">
                <Col xs={{ span: 12, order: 1 }} lg={{ span: 6, order: 1 }}>
                    <Row>
                        <Col {...keyColsProps} className={`${columnKeySize} mc-driver-details-card__key`}>{messages.dateOfBirth}</Col>
                        <Col {...valueColsProps} className={`${columnValueSize} mc-driver-details-card__value`}>
                            {dayjs(driver.dateOfBirth).format('DD/MM/YYYY')}
                        </Col>
                    </Row>
                </Col>
                <Col xs={{ span: 12, order: 4 }} lg={{ span: 6, order: 2 }}>
                    <Row>
                        <Col {...keyColsProps} className={`${columnEqualSize} mc-driver-details-card__key`}>{messages.accidence}</Col>
                        <Col {...valueColsProps} className={`${columnEqualSize} mc-driver-details-card__value`}>
                            {`${driver.claimsCountInFiveYears} ${messages.inTheLastFiveYears}`}
                        </Col>
                    </Row>
                </Col>
                <Col xs={{ span: 12, order: 2 }} lg={{ span: 6, order: 3 }}>
                    <Row>
                        <Col {...keyColsProps} className={`${columnKeySize} mc-driver-details-card__key`}>{messages.occupation}</Col>
                        <Col {...valueColsProps} className={`${columnValueSize} mc-driver-details-card__value`}>
                            {translator(`typekey.OccupationType_Ext.${driver.occupationKey}`)}
                        </Col>
                    </Row>
                </Col>
                {!licenceValidated && (
                    <Col xs={{ span: 12, order: 5 }} lg={{ span: 6, order: 4 }}>
                        <Row>
                            <Col {...keyColsProps} className={`${columnEqualSize} mc-driver-details-card__key`}>{messages.convictions}</Col>
                            <Col {...valueColsProps} className={`${columnEqualSize} mc-driver-details-card__value`}>
                                {driver.convictionsCountInFiveYears > 0
                                    ? `${driver.convictionsCountInFiveYears} ${messages.inTheLastFiveYears}`
                                    : messages.none}
                            </Col>
                        </Row>
                    </Col>
                )}
                <Col xs={{ span: 12, order: 3 }} lg={{ span: 6, order: 5 }}>
                    <Row>
                        <Col {...keyColsProps} className={`${columnKeySize} mc-licence-held-years mc-driver-details-card__key`}>{messages.drivingLicence}</Col>
                        <Col {...valueColsProps} className={`${columnValueSize} mc-driver-details-card__value custom-padding-right`}>
                            {licenceValidated ? translator(`typekey.LicenceType_Ext.${driver.licenceTypeKey}`) : (`${translator(`typekey.LicenceType_Ext.${driver.licenceTypeKey}`)},
                            held for ${translator(`typekey.LicenceHeld_Ext.${driver.licenceHeldForKey}`)}`)}
                        </Col>
                    </Row>
                </Col>
            </Row>
        </HDPlaceholderWithHeader>
    );
};

HDDriverDetails.propTypes = {
    driver: PropTypes.shape({
        id: PropTypes.string,
        name: PropTypes.string,
        role: PropTypes.string,
        dateOfBirth: PropTypes.instanceOf(Date),
        licenceTypeKey: PropTypes.string,
        licenceHeldForKey: PropTypes.string,
        occupationKey: PropTypes.string,
        claimsCountInFiveYears: PropTypes.number,
        convictionsCountInFiveYears: PropTypes.number
    }).isRequired
};

export default HDDriverDetails;
