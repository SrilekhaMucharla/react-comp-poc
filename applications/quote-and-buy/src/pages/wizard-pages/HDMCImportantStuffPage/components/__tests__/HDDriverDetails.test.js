import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import { useSelector, Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import Adapter from 'enzyme-adapter-react-16';
import HDDriverDetails from '../HDDriverDetails';

Enzyme.configure({ adapter: new Adapter() });
const middlewares = [];
const mockStore = configureStore(middlewares);
let store;
const mockAppState = {
    wizardState: {
        app: {
            driverLicenceDetails: []
        },
    },
};
jest.mock('react-redux', () => ({
    ...jest.requireActual('react-redux'),
    useSelector: jest.fn(),
}));

describe('<HDDriverDetails />', () => {
    beforeEach(() => {
        useSelector.mockImplementation((callback) => {
            return callback(mockAppState);
        });
        store = mockStore({});
    });
    afterEach(() => {
        useSelector.mockClear();
    });
    test('render with mandatory props', () => {
        const wrapper = shallow(
            <Provider store={store}>
                <HDDriverDetails driver={{
                    dateOfBirth: '18/11/1980',
                    occupation: 'Teacher',
                    drivingLicence: 'Full UK Manual, held for 16 years',
                    accidence: '1 in the last 5 years',
                    convictions: 'None'
                }} />
            </Provider>
        );
        expect(wrapper).toMatchSnapshot();
    });
});
