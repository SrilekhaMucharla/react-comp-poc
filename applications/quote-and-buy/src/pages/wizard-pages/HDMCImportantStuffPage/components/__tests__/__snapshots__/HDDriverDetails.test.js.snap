// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HDDriverDetails /> render with mandatory props 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "notify": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": null,
      },
    }
  }
>
  <HDDriverDetails
    driver={
      Object {
        "accidence": "1 in the last 5 years",
        "convictions": "None",
        "dateOfBirth": "18/11/1980",
        "drivingLicence": "Full UK Manual, held for 16 years",
        "occupation": "Teacher",
      }
    }
  />
</ContextProvider>
`;
